src/attributes.md

```diff
@@ -87,11 +82,33 @@ fn some_unused_variables() {
 }
 ```
 
-There are three kinds of attributes:
+## Meta Item Attribute Syntax
 
-* Built-in attributes
-* Macro attributes
-* Derive mode helper attributes
+A "meta item" is the syntax used for the _Attr_ rule by built-in attributes
```

Comment:
"built-in attributes" -> "most of built-in attributes", this is not a requirement but rather a "tradition" and implementation detail.
`cfg_attr` for example, doesn't fit into the meta item syntax and may contain attributes with "unrestricted" input inside it.

Also, `meta` macro fragment not accepting attributes with the unrestricted input is generally a bug / something under-implemented, but I guess the reference needs to describe the actual compiler behavior rather than the intent.

Reply:
Ah, good point. If it changes, I'll try to make sure this gets updated. 

My plan is to expand this section in the future to include some of the common templates you added (like _MetaNameValueStr_) so that other places in the documentation can just say something like "`crate_name` uses the _MetaNameValueStr_ syntax to …". When I do that, I might tweak the wording here to make the relationship to built-in attributes clearer.